presto-main/src/main/java/com/facebook/presto/operator/aggregation/MapUnionSumAggregation.java

```diff
@@ -0,0 +1,161 @@
+import com.facebook.presto.common.type.MapType;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.TypeSignatureParameter;
+import com.facebook.presto.common.type.TypeUtils;
+import com.facebook.presto.metadata.BoundVariables;
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.metadata.SqlAggregationFunction;
+import com.facebook.presto.operator.aggregation.AggregationMetadata.AccumulatorStateDescriptor;
+import com.facebook.presto.operator.aggregation.state.MapUnionSumState;
+import com.facebook.presto.operator.aggregation.state.MapUnionSumStateFactory;
+import com.facebook.presto.operator.aggregation.state.MapUnionSumStateSerializer;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.INPUT_CHANNEL;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.spi.StandardErrorCode.EXPECTING_NUMERIC_TYPE;
+import static com.facebook.presto.spi.function.Signature.numericTypeParameter;
+import static com.facebook.presto.spi.function.Signature.typeVariable;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class MapUnionSumAggregation
+        extends SqlAggregationFunction
+{
+    public static final MapUnionSumAggregation MAP_UNION_SUM = new MapUnionSumAggregation();
+    public static final String NAME = "map_union_sum";
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(MapUnionSumAggregation.class, "output", MapUnionSumState.class, BlockBuilder.class);
+    private static final MethodHandle INPUT_FUNCTION = methodHandle(MapUnionSumAggregation.class, "input", Type.class, Type.class, MapUnionSumState.class, Block.class);
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(MapUnionSumAggregation.class, "combine", MapUnionSumState.class, MapUnionSumState.class);
+
+    public MapUnionSumAggregation()
+    {
+        super(NAME,
+                ImmutableList.of(numericTypeParameter("K"), typeVariable("V")),
+                ImmutableList.of(),
+                parseTypeSignature("map<K,V>"),
+                ImmutableList.of(parseTypeSignature("map<K,V>")));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Aggregate all the maps into a single map";
```

Comment:
This description doesn't have the sum logic.

Reply:
fixed